### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#1331] [Q&A] [answered]
If I have some data like below, how would I have Visidata create a sheet listing the students with both the highest and lowest score for each test (a column each for min & max)?  This should work for a larger dataset with more tests and students.            

I was thinking it would be possible with the Frequency sheet and aggregates of min and max. I can't figure out how to also show columns from the rows that match the min/max values so I can include the student name.  If the number of tests or students goes up, it would scale well.

This seems similar to SQL window functions, https://en.wikipedia.org/wiki/Window_function_(SQL).  Visidata doesn't support a feature like window functions.

student,test,score
a,1,80          
b,1,90          
c,1,100         
a,2,90          
b,2,90          
c,2,85          
a,3,95          
b,3,90          
c,3,93

**Comment by saulpw** [comment, marked as answer]
The `keymax` aggregator will provide the key of the highest value.  Then you'd need a corresponding `keymin` aggregator (not yet impl).

**Comment by frosencrantz** [comment]
Ah! The keymax aggregator was not very clear to me.  For my sample data I would make the student column a key column, and then add "min max keymax" as the aggregators to the score column, and create a frequency sheet from the test column.

And file a wishlist for keymin...  😃 

Thanks @saulpw !
